deep_quoridor/src/osaz/alphazero_quoridor.py

```diff
@@ -8,6 +8,25 @@
 import getpass
 import json
 import os
+import sys
+import multiprocessing
+
+# Set TF environment variables for GPU memory management and device visibility.
+# This must be done BEFORE TensorFlow is imported.
+os.environ['TF_FORCE_GPU_ALLOW_GROWTH'] = 'true'
+os.environ['CUDA_VISIBLE_DEVICES'] = '0'
```

Comment:
Do you know if this is necessary? I think just leaving this env var undefined allows TF to use whatever GPU devices it finds, right?

Reply:
I just tried it without it and it worked for me. I also had hang thread sometimes when running without GPU (CPU only) so I don't think those hangs are related to the GPU patch. I also agree the right way to deal with those is to kill the runaway processes.